### PR TITLE
Problem solutions PR

### PR DESCRIPTION
**Pull request by mertkantaroglu**
Hello,

In this branch me and @josuerushanika;

- Created Solver class in solver.rb file.
- Created solver_spec.rb file for tests.
- Created tests for fizzbuzz, reverse string and factorial methods.
- Created factorial method that takes one argument, an integer N, and returns the [factorial] for that number. 
- Created reverse string that takes one argument, a string word, and returns word reversed.
- Created fizzbuzz method that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
  - When N is divisible by 3, return "fizz".
  - When N is divisible by 5, return "buzz".
  - When N is divisible by 3 and 5, return "fizzbuzz".
  - Any other case, return N as a string (e.g. say N is 7 then return "7").
- Fixed linter errors

Looking forward to hearing from you.

Thanks.

### CODE REVIEW

File: spec/factorial_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      expect(solver.factorial(5)).to eq(120)
+    end
+
+    it 'factorial 0 should be 1' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'factorial 1 should be 1' do
+      expect(solver.factorial(1)).to eq(1)
+    end
```
Comment (nahnahdev) [inline spec/factorial_spec.rb:21]: - [x] Great job so far @mertkantaroglu and @josuerushanika However I noticed that you're missing one of the tests under the `#factorial` test cases. Per the project requirements below, on the last line, there needs to also be a test that handles a negative value which raises an exception as an expectation 😉 
- [x] So kindly include that in your tests as well 💯 

> Create a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @mertkantaroglu and @josuerushanika , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Tests well written and passing ✔️ 
- No linter errors ✔️ 
- Good use of GitFlow ✔️ 
- Rspec used for testing ✔️ 
- Professional README ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/factorial_spec.rb (hunk 1 above)
Comment (mertkantaroglu) [reply spec/factorial_spec.rb:21]: Thank you for the heads up @NahnahAJ

**Review by tugsem — APPROVED** [review]
## STATUS: APPROVED 👍🎯  
Hello @mertkantaroglu  and @josuerushanika 👋 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ correct use of gitflow
✔️ linters are passing
✔️ professional documentation
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mertkantaroglu** [comment]
Thank you @tugsem 👍
